opal/runtime/opal_progress.c

```diff
@@ -224,13 +311,15 @@ int opal_progress(void)
         events += (callbacks[i])();
     }
 
-    /* Run low priority callbacks and events once every 8 calls to opal_progress().
+    /* Run low priority callbacks and events once every <N> calls to opal_progress().
      * Even though "num_calls" can be modified by multiple threads, we do not use
      * atomic operations here, for performance reasons. In case of a race, the
      * number of calls may be inaccurate, but since it will eventually be incremented,
      * it's not a problem.
+     * N = 1024 if opal_async_progress_thread_spawned, otherwise N = 8.
      */
-    if (((num_calls++) & 0x7) == 0) {
+    const uint32_t mod = opal_async_progress_thread_spawned ? 0x3FF : 0x7;
```

Comment:
that is very low priority for lp callbacks, especially taking into account that you enable yield for the progress thread. What's the rationale ?

Reply:
I just picked an arbitrary bigger number than the current value (8), there is not really heuristic about 1024. If you think of another value I'll be eager to take it.

Reply:
7 was also arbitrary, but at least that gave our event library the opportunity to do some progress from time to time. From that perspective it looks to me as 1024 is a little to long, let's go for something around 256, and will adapt from there if we notice it takes away too many resources from the communication progress.

That being said, in a progress thread it would make sense to have the low priority queue tested on a time-base interval.

Reply:
set to 256